Bearded.Utilities.Tests/Core/Random/SharedTests.cs

```diff
@@ -0,0 +1,441 @@
+            }
+        }
+        
+        public abstract class TheIntMethod : BaseTests.RandomMethodWithoutParameters<int>
+        {
+            [Property]
+            public void ShouldReturnDifferentValues(int seed)
+            {
+                Seed(seed);
+                var first = CallMethod();
+
+                sequence(100, CallMethod)
+                    .Should().Contain(i => i != first);
+            }
+        }
+
+        public abstract class TheIntWithMaxMethod : BaseTests.RandomMethodWithOneParameter<int>
+        {
+            [Property]
+            public void ShouldReturnValueInRange(int seed, PositiveInt max)
+            {
+                ResultOfCallingWith(seed, max.Get)
+                    .Should().BeInRange(0, max.Get);
+            }
+
+            [Property]
+            public void ShouldThrowForNegativeMaximum(int seed, NegativeInt max)
+            {
+                CallingWith(seed, max.Get).Should().Throw<ArgumentException>();
+            }
+
+            [Property]
+            public void ShouldReturnZeroForZeroMaximum(int seed)
+            {
+                ResultOfCallingWith(seed, 0).Should().Be(0);
+            }
+        }
+
+        public abstract class TheIntWithMinAndMaxMethod : BaseTests.RandomMethodWithTwoParameters<int>
+        {
+            [Property]
+            public void ShouldReturnValueInRange(int seed, int min, int max)
+            {
+                if (max < min)
+                    (min, max) = (max, min);
+
+                ResultOfCallingWith(seed, min, max)
+                    .Should().BeInRange(min, max);
+            }
+
+            [Property]
+            public void ShouldThrowIfMaximumLessThanMinimum(int seed, int min, int max)
+            {
+                if (max == min)
+                    max--;
+                if (max > min)
+                    (min, max) = (max, min);
+
+                CallingWith(seed, min, max)
+                    .Should().Throw<ArgumentException>();
+            }
+
+            [Property]
+            public void ShouldReturnMinimumIfMinimumEqualsMaximum(int seed, int minMax)
+            {
+                ResultOfCallingWith(seed, minMax, minMax)
+                    .Should().Be(minMax);
+            }
+        }
+        
+        public abstract class TheLongMethod : BaseTests.RandomMethodWithoutParameters<long>
+        {
+            [Property]
+            public void ShouldReturnDifferentValues(int seed)
+            {
+                Seed(seed);
+                var first = CallMethod();
+
+                sequence(100, CallMethod)
+                    .Should().Contain(i => i != first);
+            }
+        }
+
+        public abstract class TheLongWithMaxMethod : BaseTests.RandomMethodWithOneParameter<long>
+        {
+            [Property]
+            public void ShouldReturnValueInRange(int seed, long max)
+            {
+                max = Math.Max(1, max);
```

Comment:
`max = max == 0 ? 1 : Math.Abs(max)` would provide wider coverage of test cases for the same number of runs.

Reply:
And throw when `max == long.MinValue`. I don't think it's worth it.